io/driver/module_unload_load.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Naresh Bannoth<nbannoth@in.ibm.com>
+#
+
+import time
+import os
+from avocado import main
+from avocado.utils import process
+from avocado.utils import linux_modules, genio
+from avocado.utils import pci
+from avocado import Test
+
+
+class ModuleLoadUnload(Test):
+
+    """
+    This test is for driver module unload and load verification.
+    :param module: Name of the driver module
+    :param iteration: Number of time to unload and load the module
+    :only_io True for single provide module and False for All pci modules
+    """
+    def setUp(self):
+        """
+        get parameters.
+        """
+        self.module = self.params.get('module', default=None)
+        self.iteration = self.params.get('iteration', default=1)
+        self.only_io = self.params.get('only_io', default=None)
+        # self.fc = self.params.get('fc', default=None)
+        self.load_unload_sleep_time = 30
+        self.multipath_sleep_time = 5
```

Comment:
@Naresh-ibm recently, I had faced similar issue of removing multipath before reloading module, and handled it via:
```
# cat module_reload.sh 
#!/usr/bin/bash

for i in {1..20}
do
    while [[ `lsmod | grep ^mpt3sas | awk '{print $NF}'` != '0' ]]
    do
        multipath -F
        sleep 5
    done
        lsmod | grep ^mpt3sas
    echo "Iteration $i"
    echo "------------"
    rmmod mpt3sas && sleep 60
    modprobe mpt3sas && sleep 300
    dmesg -c
    echo "============================================"
done
```
Please see if this approach is helpful here

Reply:
or use `wait_for()`